java/src/main/java/ai/rapids/cudf/CudaException.java

```diff
@@ -30,9 +30,168 @@
 public class CudaException extends RuntimeException {
   CudaException(String message) {
     super(message);
+    isFatal = message.startsWith("Fatal");
```

Comment:
Ideally we should not be scraping messages or adding special members to distinguish types of exceptions.  IMO there should be a specific class in Java for fatal CUDA exceptions that is a subclass of `CudaException`.  That way an application can trivially catch all types of CUDA exceptions by catching `CudaException` but can also trivially only catch fatal CUDA exceptions by catching `CudaFatalException`.

This also has the benefit of not needing an `isFatal` method that a caller would need to call, and it also makes it painfully obvious to the user in the stacktrace that this is a fatal CUDA exception rather than just any CUDA exception because the specific class name will appear in the stacktrace.  Otherwise the user will have to discern by interpreting the message.

Message scraping in exceptions is an anti-pattern.  It indicates there isn't sufficient resolution in the exception types to catch the desired subclass of exceptions directly.

Reply:
I replaced `isFatal` with `CudaFatalException`. However, we still need to scrape the message to get the error name. Because `env->ThrowNew` doesn't accept argument other than error message.

Reply:
We don't need to scrape the error messages.  The exception instance with extra custom arguments can be created explicitly in the JNI and then `env->Throw` can be used to throw it.  The JNI code can lookup the class, lookup the `<init>` method on the class, call `env->NewObject` with the construction method and extra error code to construct the exception object, then call `env->Throw` with the exception object.  Or you could have the JNI lookup and call a static method, passing the error message and error code, and let the static method throw the appropriate exception type.  Bottom line is there are multiple ways to avoid scraping messages, and we should leverage one of them here.

Reply:
I applied the previous approach: `env->Throw` with `env->NewObject`. Thank you for such a detailed recommendation.